lib/utils/disk.go

```diff
@@ -37,3 +43,39 @@ func PercentUsed(path string) (float64, error) {
 	ratio := float64(stat.Blocks-stat.Bfree) / float64(stat.Blocks)
 	return Round(ratio * 100), nil
 }
+
+// CanUserWriteTo attempts to check if a user has write access to certain path.
+// It also works around the program being run as root and tries to check
+// the permissions of the user who executed the program as root.
+// This should only be used for string formatting or inconsequential use cases
+// as it's not bullet proof and can report wrong results.
+func CanUserWriteTo(path string) bool {
+	// prevent infinite loops with a maxIterations
+	maxIterations := 10
+	for {
+		if maxIterations == 0 {
+			return false
+		}
+		maxIterations--
+
+		_, err := os.Stat(path)
+		if err == nil {
+			break
+		}
+		if errors.Is(err, fs.ErrNotExist) {
+			path = filepath.Dir(path)
```

Comment:
I don't really love the repeated `filepath.Dir()` calls with `maxIterations` but wow is Go's stdlib really rough for path manipulation. I can't really think of a better solution here.